server/paymentProviders/transferwise/index.ts

```diff
@@ -242,13 +242,30 @@ async function createTransfer(
     throw new TransferwiseError(message, null, { quote });
   }
 
+  const country = recipient.details.address?.country;
+  let reference = `${expense.collective.slug.toUpperCase().substring(0, 3)}${expense.id}`; // MAX 10 chars for the US
+  switch (country) {
+    case 'GBP': // max 18 chars
+      reference = `${expense.collective.slug.toUpperCase().substring(0, 10)}${expense.id}`;
+      break;
+    case 'EUR': // max 35 chars
+      reference = `${expense.collective.slug.substring(0, 25)}${expense.id}`;
+      break;
+    case 'USD': // max 10 chars
+      reference = `${expense.collective.slug.toUpperCase().substring(0, 3)}${expense.id}`;
+      break;
+    case 'MXN': // max 100 chars
+      reference = `${expense.collective.slug.substring(0, 100)}${expense.id}`;
+      break;
+  }
```

Comment:
The substring should always be `substring(0, maxLength - expense.id.toString().length)`. Looks like this was considered for some (GBP, EUR, USD) but not MXN

Reply:
Yes, it's also wrong to use the currency code instead of country code. Please consider my PRs more as a way to kickstart the conversation. Feel free to rewrite (lmk if you prefer that I just write a github issue).